lib/ansible/plugins/action/__init__.py

```diff
@@ -549,7 +563,7 @@ def _execute_module(self, module_name=None, module_args=None, tmp=None, task_var
 
         # Fix permissions of the tmp path and tmp files.  This should be
         # called after all files have been transferred.
-        self._fixup_perms(tmp, remote_user, recursive=True)
+        self._fixup_perms(tmp, remote_user, recursive=True, execute=is_binary)
```

Comment:
We execute the interpreter directly by substituting ansible_<interpreter>_interpreter so not sure that this is needed (unless we want to change to always execute the file?